### DISCUSSION TITLE
plane normal

### DISCUSSION THREAD

**Discussion by srraju** [mtex-toolbox/mtex#533] [Ask Anything]
Hi,

I know its a simple question. But wondering how to find a plane normal vector (or diffraction vector, g) using mtex.
According to Partridge, it is [h k i 3/2 (a/c)^2l] for (hkil).
From some literature, i found that [50-52] is normal to (10-11) plane.
and [30-32] for (10-12).
 I am confused how these values round up.

Thanks,
Raju

**Comment by ralfHielscher** [comment]
Would you like to tell us what you really want to achieve?

Ralf.

**Reply by srraju** [reply]
Hi Ralf,

Thanks for your prompt reply.
I am calculating the contrast factors for X-ray diffraction analysis. It requires the plane normal (or diffraction vector, g) to be computed. 
![image](https://user-images.githubusercontent.com/29395354/96426727-5604cd80-1249-11eb-803a-ab6cb5005e86.png)
The equation above shows modified Warren-Averbach analysis in which the computation of contrast factor, Chkl requires the diffraction vector, g.

Ivan et al. computed the reduced integer forms of the diffraction vector (see attached snaphshot). Since mtex has many capabilities for orientation analysis, i am checking if i can find a simple solution through mtex to verify these.
![image](https://user-images.githubusercontent.com/29395354/96427104-e216f500-1249-11eb-8a01-b29afa572727.png)

[Ivanov2020_Article_ApplicationOfDifferentDiffract.pdf](https://github.com/mtex-toolbox/mtex/files/5400673/Ivanov2020_Article_ApplicationOfDifferentDiffract.pdf)

Hope my explanation is clear.
Looking for ward for the reply.

Thanks,
Raju

**Reply by srraju** [reply]
Calculator 7 and 8 in the following link does calculate plane normal, but the computed vectors are in decimal forms but not the simplest integer forms.
http://ssd.phys.strath.ac.uk/resources/crystallography/crystallographic-direction-calculator/

**Comment by ralfHielscher** [comment]
I guess what you are looking for is
```
abc = [1 1 3]
cs = crystalSymmetry('6/mmm',abc)
m = Miller(1,0,-1,0,'hkl',cs)
m.dispStyle = 'uvw'
round(m)
```
Ralf.
